test/e2e/fixture-builder.js

```diff
@@ -466,111 +466,171 @@ class FixtureBuilder {
   }
 
   withPermissionControllerConnectedToTestDapp({
-    restrictReturnedAccounts = true,
     account = '',
     useLocalhostHostname = false,
   } = {}) {
     const selectedAccount = account || DEFAULT_FIXTURE_ACCOUNT;
-    return this.withPermissionController({
-      subjects: {
-        [useLocalhostHostname ? DAPP_URL_LOCALHOST : DAPP_URL]: {
-          origin: useLocalhostHostname ? DAPP_URL_LOCALHOST : DAPP_URL,
-          permissions: {
-            eth_accounts: {
-              id: 'ZaqPEWxyhNCJYACFw93jE',
-              parentCapability: 'eth_accounts',
-              invoker: DAPP_URL,
-              caveats: restrictReturnedAccounts && [
-                {
-                  type: 'restrictReturnedAccounts',
-                  value: [
-                    selectedAccount.toLowerCase(),
-                    '0x09781764c08de8ca82e156bbf156a3ca217c7950',
-                    ERC_4337_ACCOUNT.toLowerCase(),
-                  ],
+    const subjects = {
+      [useLocalhostHostname ? DAPP_URL_LOCALHOST : DAPP_URL]: {
+        origin: useLocalhostHostname ? DAPP_URL_LOCALHOST : DAPP_URL,
+        permissions: {
+          'endowment:caip25': {
+            caveats: [
+              {
+                type: 'authorizedScopes',
+                value: {
+                  requiredScopes: {},
+                  optionalScopes: {
+                    'eip155:1337': {
+                      accounts: [
+                        `eip155:1337:${selectedAccount.toLowerCase()}`,
+                      ],
+                    },
+                  },
+                  isMultichainOrigin: false,
                 },
-              ],
-              date: 1664388714636,
-            },
+              },
+            ],
+            id: 'ZaqPEWxyhNCJYACFw93jE',
+            date: 1664388714636,
+            invoker: DAPP_URL,
+            parentCapability: 'endowment:caip25',
           },
         },
       },
+    };
+    return this.withPermissionController({
+      subjects,
     });
   }
 
-  withPermissionControllerSnapAccountConnectedToTestDapp(
-    restrictReturnedAccounts = true,
-  ) {
-    return this.withPermissionController({
-      subjects: {
-        [DAPP_URL]: {
-          origin: DAPP_URL,
-          permissions: {
-            eth_accounts: {
-              id: 'ZaqPEWxyhNCJYACFw93jE',
-              parentCapability: 'eth_accounts',
-              invoker: DAPP_URL,
-              caveats: restrictReturnedAccounts && [
-                {
-                  type: 'restrictReturnedAccounts',
-                  value: ['0x09781764c08de8ca82e156bbf156a3ca217c7950'],
+  withPermissionControllerConnectedToTestDappWithTwoAccounts() {
```

Comment:
Ah, I see at least this might be needed. I'm guessing that when `restrictReturnedAccounts` is `false`  for `withPermissionControllerConnectedToTestDapp`, it allows the dapp to access all accounts? I don't quite follow what was happening in that situation.

Maybe this is another change we can make in a separate PR

Reply:
Done here https://github.com/MetaMask/metamask-extension/pull/29671